api/submission/fetch_course_report_test.go

```diff
@@ -0,0 +1,110 @@
+package submission
+
+import (
+	"reflect"
+	"testing"
+
+    "github.com/edulinq/autograder/common"
+	"github.com/edulinq/autograder/api/core"
+	"github.com/edulinq/autograder/model"
+	"github.com/edulinq/autograder/report"
+	"github.com/edulinq/autograder/util"
+)
+
+func TestCourseReport(test *testing.T) {
+    testCases := []struct{
+        role model.UserRole
+        courseId string 
+        coursError bool
+        permError bool
+        result *FetchCourseReportResponse
+    }{
+        // Admin Valid Course
+        {model.RoleAdmin,"course101",false,false,expected},
+
+        // Admin Invalid course
+        {model.RoleAdmin,"course102",true,false,expected},
```

Comment:
The code you created does not deal with invalid courses (that is dealt with before your code is called by the API infrastructure).
So, that is not your responsibility to test.
In fact, we explicitly do not want you to test things that do not cover your purview.